src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java

```diff
@@ -46,78 +47,85 @@ public class ParserPDF {
-
         for (int page = document.getNumberOfPages() - PAGES_FOR_PARSE; page < document.getNumberOfPages(); page++) {
-            final String prePage = pageOCRPreparator.preparePage(document, page);
-            if (tableRecognizer.isTableExistOnPage(prePage)) {
-                log.debug("Found table on page " + page);
-                List<double[]> lines = rectangleDetector.findVerticalLinesWithOpenCV(prePage);
-                log.debug("Quantity of lines = " + lines.size());
-                List<double[]> tablesLines = tableDetector.detectQuantityOfTables(lines);
-                rectangleDetector.saveIMageWithVerticalLines2(tablesLines);
-                String fileTableName = tableRecognizer.detectTable(prePage);
-                table = rectangleDetector.detectRectangles(fileTableName);
-                table = extractTextFromScannedDocument(fileTableName);
-                dataRecognizer.recognizeLotPDFResult(table);
-                obj.put("fileName", file.getOriginalFilename());
-                StringBuilder builder = new StringBuilder();
-                for (Row row : table) {
-                    for (Column column : row.getColumns()) {
-                        builder.append(column.getParsingResult());
+            try {
+                final String preparedPage = pageOCRPreparator.preparePage(document, page);
+                if (tableRecognizer.isTableExistOnPage(preparedPage)) {
+                    log.debug("Found table on page " + page);
+                    List<double[]> lines = rectangleDetector.findVerticalLinesWithOpenCV(preparedPage, page);
+                    log.debug("Vertical lines founded = " + lines.size());
+                    List<double[]> tablesLines = tableDetector.detectQuantityOfTables(lines);
+                    rectangleDetector.saveIMageWithVerticalLines2(tablesLines, page);
+                    String fileTableName = tableRecognizer.detectTable(preparedPage, page);
+                    table = rectangleDetector.detectRectangles(fileTableName, page);
+                    table = extractTextFromScannedDocument(fileTableName);
+                    dataRecognizer.recognizeLotPDFResult(table);
+                    obj.put("fileName", file.getOriginalFilename());
+                    StringBuilder builder = new StringBuilder();
+                    for (Row row : table) {
+                        for (Column column : row.getColumns()) {
+                            builder.append(column.getParsingResult());
+                        }
                     }
+                    obj.put("text", builder);
+                } else {
+                    log.debug("Table did not found on page " + page);
                 }
-                obj.put("text", builder);
-            } else {
-                log.debug("Table did not found on page " + page);
+            } catch (RuntimeException ex) {
+                ex.printStackTrace();
+                ex.getCause();
```

Comment:
add a dedicated exception for it with a cause and a message

Reply:
Done.